src/bp/core/modules/config-reader.ts

```diff
@@ -172,21 +182,56 @@ export class ConfigReader {
     return config
   }
 
-  @Memoize()
+  private async getCachedOrFresh(key, getFresh: { fn: (...args) => any; args: any[] }) {
+    let config
+    if (!this.moduleConfigCache.has(key)) {
+      config = await getFresh.fn.call(this, ...getFresh.args)
+      this.moduleConfigCache.set(key, config)
+    } else {
+      config = this.moduleConfigCache.get(key) || {}
+    }
+    return config
+  }
+
   public async getGlobal(moduleId: string): Promise<ModuleConfig> {
-    return this.getMerged(moduleId)
+    return this.getCachedOrFresh(moduleId, { fn: this.getMerged, args: [moduleId] })
   }
 
-  // Don't @Memoize() this fn. It only memoizes on the first argument
-  // https://github.com/steelsojka/lodash-decorators/blob/master/src/memoize.ts#L15
   public getForBot(moduleId: string, botId: string, ignoreGlobal?: boolean): Promise<ModuleConfig> {
     const cacheKey = `${moduleId}//${botId}//${!!ignoreGlobal}`
-    return this.getForBotMemoized(cacheKey)
+    return this.getCachedOrFresh(cacheKey, { fn: this.getMerged, args: [moduleId, botId, yn(!!ignoreGlobal)] })
```

Comment:
Yes, I was thinking about reusability for that in the future, in case there a config that requires a different function. Changed it to be more straightforward as request